Quorum/apis/llms/llm_conversation.py

```diff
@@ -0,0 +1,105 @@
+from enum import StrEnum, auto
+from pathlib import Path
+
+class MessageRole(StrEnum):
+    USER = auto()
+    ASSISTANT = auto()
+
+
+MAX_TOKENS = 1024
+
+
+def tmp_send_message_to_claude(client, model, messages):
+    try:
+        response = client.messages.create(
+            max_tokens=MAX_TOKENS,
+            model=model,
+            messages=messages
+        )
+    except:
+        raise
+
+    with open(Path(__file__).parent / 'ClaudeExecutionsNum.txt', 'r+') as f:
+        num = int(f.read())
+        f.seek(0)
+        f.write(str(num + 1))
+        f.truncate()
+
+    return response
+
+
+class LLMConversation:
+    '''
+    Description
+    -----------
+    At least GPT and Claude share the exact same API. Probably other LLMs use the same API as well,
+    just using different clients. This abstract class builds is expected to be inherited and build
+    the relevant client to be used with the same implementation of interacting with the LLM while 
+    maintaining a conversation context.
+    '''
```

Comment:
Reading this description made me realize that since we're open source, we can provide users the option to implement whatever llm they want, so inheritance is indeed the better approach

Reply:
That's a good point but I didn't even think about that. We already discussed we may want to use a different model once GPT will be cheaper or find an open source one.